internal/locate/region_request.go

```diff
@@ -141,30 +145,60 @@ type RequestErrorStats struct {
 
 // NewRegionRequestRuntimeStats returns a new RegionRequestRuntimeStats.
 func NewRegionRequestRuntimeStats() *RegionRequestRuntimeStats {
-	return &RegionRequestRuntimeStats{
-		RPCStats: make(map[tikvrpc.CmdType]*RPCRuntimeStats),
-	}
+	return &RegionRequestRuntimeStats{}
 }
 
 // RPCRuntimeStats indicates the RPC request count and consume time.
 type RPCRuntimeStats struct {
-	Count int64
+	Cmd   tikvrpc.CmdType
+	Count uint32
 	// Send region request consume time.
-	Consume int64
+	Consume time.Duration
 }
 
 // RecordRPCRuntimeStats uses to record the rpc count and duration stats.
 func (r *RegionRequestRuntimeStats) RecordRPCRuntimeStats(cmd tikvrpc.CmdType, d time.Duration) {
-	stat, ok := r.RPCStats[cmd]
+	if r.FirstRPCStats.Count == 0 || r.FirstRPCStats.Cmd == cmd {
+		r.FirstRPCStats.Cmd = cmd
+		r.FirstRPCStats.Count++
+		r.FirstRPCStats.Consume += d
+		return
+	}
+	if r.OtherRPCStatsMap == nil {
+		r.OtherRPCStatsMap = make(map[tikvrpc.CmdType]*RPCRuntimeStats)
+	}
+	stat, ok := r.OtherRPCStatsMap[cmd]
 	if !ok {
-		r.RPCStats[cmd] = &RPCRuntimeStats{
+		r.OtherRPCStatsMap[cmd] = &RPCRuntimeStats{
+			Cmd:     cmd,
 			Count:   1,
-			Consume: int64(d),
+			Consume: d,
 		}
 		return
 	}
 	stat.Count++
-	stat.Consume += int64(d)
+	stat.Consume += d
+}
+
+// GetRPCCount returns the total rpc types count.
+func (r *RegionRequestRuntimeStats) GetRPCCount() int {
```

Comment:
How about to use `GetRPCStatsCount` (or sth else)? This name make me think it returns the total number of RPCs of all cmd types.

Reply:
great, done.